modules/efk-logging-fluentd-undefined.adoc

```diff
@@ -0,0 +1,146 @@
+If you set the `MERGE_JSON_LOG` parameter to `true`, see the Note below.
+|`CDM_UNDEFINED_DOT_REPLACE_CHAR="_"`
+
+|===
+
+[NOTE]
+====
+If you set the `MERGE_JSON_LOG` parameter in the log collector daemonset and `CDM_UNDEFINED_TO_STRING` environment variables to true, you might receive an Elasticsearch *400* error. 
+The error occurs because when`MERGE_JSON_LOG=true`, the log collector adds fields with data types other than string. When you set `CDM_UNDEFINED_TO_STRING=true`, 
+the log collector attempts to add those fields as a string value resulting in the Elasticsearch 400 error. The error clears when the log collector rolls over 
+the indices for the next day’s logs
+
+When the log collector rolls over the indices, it creates a brand new index. The field definitions are updated and you will not get the *400* error.
+====
+
+.Procedure
+
+Use the `CDM_*` parameters to configure undefined and empty field processing.
+
+. Configure how to process fields, as needed:
+
+.. Specify the fields to move using `CDM_EXTRA_KEEP_FIELDS`.
+
+.. Specify any empty fields to retain in the `CDM_KEEP_EMPTY_FIELDS` parameter in CSV format.
+
+. Configure how to process undefined fields, as needed:
+
+.. Set `CDM_USE_UNDEFINED` to `true` to move undefined fields to the top-level `undefined` field:
+
+.. Specify a name for the undefined fields using the `CDM_UNDEFINED_NAME` parameter.
+
+.. Set `CDM_UNDEFINED_MAX_NUM_FIELDS` to a value other than the default `-1`, to set an upper bound on the number of undefined fields 
+in a single record.  
+
+. Specify `CDM_UNDEFINED_DOT_REPLACE_CHAR` to change any dot `.` characters in an undefined field name to another character.
+For example, if `CDM_UNDEFINED_DOT_REPLACE_CHAR=@@@` and there is a field named `foo.bar.baz` the field is transformed into `foo@@@bar@@@baz`.
+
+. Set `UNDEFINED_TO_STRING` to `true` to convert undefined fields to their JSON string representation.  
+
+[NOTE]
+====
+If you configure the `CDM_UNDEFINED_TO_STRING` or `CDM_UNDEFINED_MAX_NUM_FIELDS` parameters, you use the `CDM_UNDEFINED_NAME` to change the undefined field name. This field is needed because `CDM_UNDEFINED_TO_STRING` or `CDM_UNDEFINED_MAX_NUM_FIELDS` could change the value type of the undefined field. When `CDM_UNDEFINED_TO_STRING` or `CDM_UNDEFINED_MAX_NUM_FIELDS` is set to true and there are more undefined fields in a log, the value type becomes `string`. Elasticsearch stops accepting records if the value type is changed, for example, from JSON to JSON string. When `CDM_UNDEFINED_TO_STRING` or `CDM_UNDEFINED_MAX_NUM_FIELDS` are `false`, the value type of the undefined field is JSON.
+
+For example, when `CDM_UNDEFINED_TO_STRING` or `CDM_UNDEFINED_MAX_NUM_FIELDS` are set to false, the value type is `json`. If you change `CDM_UNDEFINED_MAX_NUM_FIELDS` to `true` and there are more undefined fields in a log, the value type becomes `string` (json string). Elasticsearch stops accepting records if the value type is changed.
```

Comment:
Please note the value of `CDM_UNDEFINED_MAX_NUM_FIELDS` should be a number, and could not be `true` or `false`

Reply:
@QiaolingTang 
Should this state: 
_For example, when `CDM_UNDEFINED_TO_STRING` is `false` or `CDM_UNDEFINED_MAX_NUM_FIELDS` is default, `-1`, the value type is `json`. If you change `CDM_UNDEFINED_MAX_NUM_FIELDS` to to a value other than default...